client/src/pages/Signup.js

```diff
@@ -0,0 +1,155 @@
+  const [password, setPassword] = useState('')
+  const [open, setOpen] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('')
+
+  const createAccount = async (event) => {
+    try {
+      event.preventDefault()
+      let passedFields = true
+
+      //error handling here for fields
+      if (!validateEmail(email)) {
+        setErrorMessage('Please enter a valid email')
+        passedFields = false
+
+      }
+      if (password.length < 7) {
+        setErrorMessage('Please enter a password longer than 7 characters')
+        passedFields = false
+      }
+      if (email === '' || company === '' || password === '') {
+        setErrorMessage('Please fill in all fields')
+        passedFields = false
+      }
+      if (!passedFields) {
+        setOpen(true)
+        return
+      }
+
+      //if frontend validations pass, make server call here to create user
+      const res = await axios.get('welcome', {
+        email: email,
+        password: password,
+        company: company
+      })
+      if (res.status === 400) {
+        setErrorMessage('That email already exists. Please choose another')
+        setOpen(true)
+        return
+      }
+      if (res.data.user) {
+        const { history } = props
+        history.push('/main')
+      }
+
+    } catch (error) {
+      console.error(error)
+    }
+  }
+  return (
+    <Container className={classes.mainContainer} component="main" maxWidth="xs">
+      <CssBaseline />
+      <div className={classes.paper}>
+        {loginSignup === '/signup' ?
+          <React.Fragment>
+            <Typography variant="h1">
+              Lets Get Started!
+        </Typography>
+            <Typography >
+              Create an account
+        </Typography>
+          </React.Fragment>
+          :
+          <React.Fragment>
+            <Typography variant="h1">
+              Welcome Back!
+        </Typography>
+          </React.Fragment>
+        }
+        <form className={classes.form} onSubmit={createAccount} noValidate>
+          <CustomTextField
+            id="email"
+            label="Email Address"
+            name="email"
+            autoComplete="email"
+
+            onChange={event => { setEmail(event.target.value) }}
+          />
+          {loginSignup === '/signup' ?
+            <CustomTextField
+              name="CompanyName"
+              label="Company Name"
+              id="CompanyName"
+              onChange={event => { setCompany(event.target.value) }}
+            />
+            :
+            ''
+          }
+          <CustomTextField
+            name="password"
+            label="Password"
+            type="password"
+            id="password"
+            onChange={event => { setPassword(event.target.value) }}
+          />
+          <SubmitButton>
+            {loginSignup === '/signup' ? 'Create' : 'Log In'}
+          </SubmitButton>
+        </form>
+      </div>
+      <Snackbar
```

Comment:
The position of this Snackbar is at the right-bottom corner now, maybe it is better to place it above the textfields.

Reply:
Something like this?

<img width="539" alt="Screen Shot 2020-07-02 at 7 43 58 PM" src="https://user-images.githubusercontent.com/47200207/86417896-701b9f80-bc9c-11ea-9a51-0c12fbe92233.png">

Reply:
I am not sure. I think some websites put the error display between "Create an account" and the textfields, some others put it between the textfields and the "Create" button. What do you think? @grogsy

Reply:
material ui textfields have an option to display errors at the bottom of the input. If this looks better I should be able to switch over:
<img width="525" alt="Screen Shot 2020-07-02 at 8 00 08 PM" src="https://user-images.githubusercontent.com/47200207/86418636-c8ec3780-bc9e-11ea-8bd0-c8dafecba0c2.png">

Reply:
Yeah I think this is better~

Reply:
I kind of agree too. Ok I can switch them